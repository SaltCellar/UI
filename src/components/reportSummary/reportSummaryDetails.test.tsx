import { shallow } from 'enzyme';
import React from 'react';
import {
  ReportSummaryDetails,
  ReportSummaryDetailsProps,
} from './reportSummaryDetails';

const props: ReportSummaryDetailsProps = {
  report: { data: [], total: { value: 100, units: 'USD' } },
  label: 'label',
  description: 'description',
  formatValue: jest.fn(() => 'formatedValue'),
};

test('report total is formated', () => {
  const view = shallow(<ReportSummaryDetails {...props} />);
  expect(props.formatValue).toBeCalledWith(props.report.total.value);
  expect(view).toMatchSnapshot();
});

test('defaults value if report is not present', () => {
  const view = shallow(<ReportSummaryDetails {...props} report={null} />);
  expect(props.formatValue).not.toBeCalled();
  expect(view).toMatchSnapshot();
});

test('defaults value if report.total is not present', () => {
  const view = shallow(
    <ReportSummaryDetails
      {...props}
      report={{ ...props.report, total: null }}
    />
  );
  expect(props.formatValue).toBeCalledWith(0);
  expect(view).toMatchSnapshot();
});
