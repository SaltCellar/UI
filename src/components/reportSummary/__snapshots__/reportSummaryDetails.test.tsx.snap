// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaults value if report is not present 1`] = `
<div
  className="reportSummaryDetails_1kpzzqo"
>
  <div
    className="value_1uvf090"
  >
    ----
  </div>
  <div
    className="text_m522r3"
  >
    <div>
      label
    </div>
  </div>
</div>
`;

exports[`defaults value if report.total is not present 1`] = `
<div
  className="reportSummaryDetails_1kpzzqo"
>
  <div
    className="value_1uvf090"
  >
    0
  </div>
  <div
    className="text_m522r3"
  >
    <div>
      label
    </div>
  </div>
</div>
`;

exports[`report total is formated 1`] = `
<div
  className="reportSummaryDetails_1kpzzqo"
>
  <div
    className="value_1uvf090"
  >
    formatedValue
  </div>
  <div
    className="text_m522r3"
  >
    <div>
      label
    </div>
  </div>
</div>
`;
